shared/utility/vision/fourcc.hpp

```diff
@@ -6,11 +6,13 @@
 
 namespace utility::vision {
 
+    // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays)
     inline constexpr uint32_t fourcc(const char (&c)[5]) {
         return c[3] << 24 | c[2] << 16 | c[1] << 8 | c[0];
     }
 
     inline std::string fourcc(const uint32_t& v) {
+        // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays)
```

Comment:
I think this section can just be replaced with an initialiser list for a string.

return {...};